File: Anul II/Medii de proiectare si programare/Laborator/lab05/Ubb.BikeContest.Service/src/IRaceService.cs
using Ubb.BikeContest.Model;
using Ubb.BikeContest.Model.data;

namespace Ubb.BikeContest.Service;

public interface IRaceService : IService<long, Race>
{
    
    Race GetRaceByName(string name);

    IEnumerable<RaceEntry> GetEntriesByRace(long raceId);

    void DeleteByIds(long participantId, long raceId);

    IEnumerable<Race> GetRacesByEngineCapacity(int engineCapacity);

    IEnumerable<RaceDto> GetRacesWithParticipantCount();

    void SaveRaceEntry(RaceEntry newEntity);

    IEnumerable<Race> GetRacesWhereNotRegisteredAndEngineCapacity(long participantId, int engineCapacity);
}

File: Anul II/Medii de proiectare si programare/Laborator/lab10/services/src/main/java/com/ubb/service/IRaceService.java
package com.ubb.service;

import com.ubb.model.Race;
import com.ubb.model.RaceEntry;
import com.ubb.dto.RaceDTO;

import java.util.Collection;
import java.util.Optional;

public interface IRaceService extends IService<Long, Race> {
    Optional<Race> getRaceByName(String name);

    Collection<RaceEntry> getEntriesByRace(Long raceID);

    void deleteByIds(Long participantID, Long raceID);

    Collection<Race> getRacesByEngineCapacity(Integer engineCapacity);

    Collection<RaceDTO> getRacesWithParticipantCount();

    void saveRaceEntry(RaceEntry newEntity);

    Collection<Race> getRacesWhereNotRegisteredAndEngineCapacity(Long participantID, Integer engineCapacity);
}
